Homework #7

BolzmanBrain opened the pull request:
- now using Spring Data
- solved N + 1 for Spring Data

BolzmanBrain committed:
Homework #7

- now using Spring Data
- solved N + 1 for Spring Data

kataus commented on homework7/src/main/java/ru/otus/spring/domain/Book.java:
```diff
@@ -0,0 +1,69 @@
+    @JoinColumn(name = "id_author")
+    private final Author author;
+
+    @ManyToOne(fetch = FetchType.EAGER)
```
Не стоит повторять значения по умолчанию, это смущает при чтении

BolzmanBrain replied:
Исправил

kataus commented on homework7/src/test/java/ru/otus/spring/services/BookCommentServiceImplTest.java:
```diff
@@ -0,0 +1,108 @@
+    @Autowired
+    private TestEntityManager tem;
+    @Autowired
+    private BookCommentServiceImpl service;
```
Получается уже интеграционный тест. ДЛя unit внешние зависимости класса должны быть замокированы

BolzmanBrain replied:
Во время лекции нам говорили, что можно использовать TestEntityManager в тестах, потому что это позволяет не обращаться к другим тестовым классам.

Этот класс здесь используется только для целей подготовки теста.
Это не внешняя зависимость тестируемого класса, а часть инфраструктуры тестирования.

kataus replied:
Ох, эти любители делать интеграционники :) Тут я не очень согласен с выводами на лекции, но не будем делать вас заложниками наших халиваров.

BolzmanBrain committed:
Homework #6

- fixes after code-review (mostly style)

BolzmanBrain committed:
Homework #7

- fixes after code-review
- fixed comments being deleted while updating a book
- removed @Transactional where it's already used by Spring Data (implementation of JpaRepository)
